StartingExample/MapScreen.cs

```diff
@@ -0,0 +1,76 @@
+﻿using GoRogue;
+using GoRogue.GameFramework;
+using GoRogue.MapGeneration;
+using GoRogue.MapViews;
+using Microsoft.Xna.Framework;
+using SadConsole;
+using XnaRect = Microsoft.Xna.Framework.Rectangle;
+
+namespace StartingExample
+{
+    class MapScreen : ContainerConsole
+    {
+        public ExampleMap Map { get; }
+        public ScrollingConsole MapRenderer { get; }
+        //Generate a map and display it.  Could just as easily pass it into
+        public MapScreen(int mapWidth, int mapHeight, int viewportWidth, int viewportHeight)
+        {
+            Map = GenerateDungeon(mapWidth, mapHeight);
+
+            // Get a console that's set up to render the map, and add it as a child of this container so it renders
+            MapRenderer = Map.CreateRenderer(new XnaRect(0, 0, viewportWidth, viewportHeight), SadConsole.Global.FontDefault);
+            Children.Add(MapRenderer);
+            Map.ControlledGameObject.IsFocused = true; // Set player to receive input, since in this example the player handles movement
+
+            // Set up to recalculate FOV and set camera position appropriately when the player moves
+            Map.ControlledGameObject.Moved += Player_Moved;
```

Comment:
This design creates a bad problem when `ControlledGameObject` changes at any moment. Since we hook the `ControlledGameObject.Moved` event, as soon as the reference changes we'll have a memory leak holding on to the `Moved` event of the old object still.

I suggest that the `Map` has an event of `ControlledGameObjectChanged` that `MapScreen` can hook to. When that happens it runs code that gets the old game object, removes the `Moved` event, and then adds the `Moved` event to the new game object.

Reply:
Whoops, good catch! This event may also be useful as part of a potential solution to the invalid cast issue.

Reply:
Note: Looking at this, It occurs to me that I'm not sure this will actually result in the memory leak described.  The result of this code is the player `Moved` event (which is effectively a delegate list) ends up holding a reference to the ` Player_Moved` function, which is defined in `MapScreen`.  If the map screen was discarded before the player, it _could_ result in the map screen failing to be garbage collected, but it shouldn't be the case the other way around, since the player is holding a reference to a function in map screen instead of the other way around.

That much said, although that leak is much less likely to occur, it's probably worth fixing up anyway, as there are plausible cases where wanting to respond to that property's assignment would be desirable, so it certainly can't hurt anything to have an event around to allow it.

Reply:
Issue should now be resolved.